src/Symfony/Component/Security/Guard/Authenticator/GuardAuthenticatorInterface.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Guard\Authenticator;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Guard\GuardAuthenticatorInterface as GuardAuthenticatorLegacyInterface;
+use Symfony\Component\Security\Guard\Token\GuardTokenInterface;
+
+/**
+ * The interface for all "guard" authenticators.
+ *
+ * The methods on this interface are called throughout the guard authentication
+ * process to give you the power to control most parts of the process from
+ * one location.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ * @author Amaury Leroux de Lens <amaury@lerouxdelens.com>
+ */
+interface GuardAuthenticatorInterface extends GuardAuthenticatorLegacyInterface
+{
+    /**
+     * Does the authenticator supports the given Request ?
+     *
+     * If this returns true, authentication will continue (e.g. getCredentials() will be called).
+     * If false, this authenticator is done. The next (if any) authenticators will be called and
+     * may authenticate the user, or leave the user as anonymous.
+     *
+     * @param Request $request
+     *
+     * @return bool
+     */
+    public function supports(Request $request);
+
+    /**
+     * Get the authentication credentials from the request and return them
+     * as any type (e.g. an associate array). If you return null, authentication
+     * will be skipped.
+     *
+     * Whatever value you return here will be passed to getUser() and checkCredentials()
+     *
+     * For example, for a form login, you might:
+     *
+     *      return array(
+     *          'username' => $request->request->get('_username'),
+     *          'password' => $request->request->get('_password'),
+     *      );
+     *
+     * Or for an API token that's on a header, you might use:
+     *
+     *      return array('api_key' => $request->headers->get('X-API-TOKEN'));
+     *
+     * @param Request $request
+     *
+     * @return mixed|null
+     */
+    public function getCredentials(Request $request);
```

Comment:
@iltar The problem with `getUser()` is that we already have a method named that - and this would cause a signature change. So we need a new method name so we can keep the old one for BC. So, it's not idea. Perhaps `getUserFromRequest()`?